executor/delete.go

```diff
@@ -158,20 +158,26 @@ func (e *DeleteExec) doBatchDelete(ctx context.Context) error {
 }
 
 func (e *DeleteExec) composeTblRowMap(tblRowMap tableRowMapType, colPosInfos []plannercore.TblColPosInfo, joinedRow []types.Datum) error {
-	// iterate all the joined tables, and got the copresonding rows in joinedRow.
+	// iterate all the joined tables, and got the corresponding rows in joinedRow.
 	for _, info := range colPosInfos {
 		if unmatchedOuterRow(info, joinedRow) {
 			continue
 		}
 		if tblRowMap[info.TblID] == nil {
-			tblRowMap[info.TblID] = kv.NewHandleMap()
+			tblRowMap[info.TblID] = kv.NewMemAwareHandleMap[[]types.Datum]()
 		}
 		handle, err := info.HandleCols.BuildHandleByDatums(joinedRow)
 		if err != nil {
 			return err
 		}
 		// tblRowMap[info.TblID][handle] hold the row datas binding to this table and this handle.
-		tblRowMap[info.TblID].Set(handle, joinedRow[info.Start:info.End])
+		_, exist := tblRowMap[info.TblID].Get(handle)
+		memDelta := tblRowMap[info.TblID].Set(handle, joinedRow[info.Start:info.End])
+		if !exist {
+			memDelta += types.EstimatedMemUsage(joinedRow, 1)
```

Comment:
1. A little question, if the handle is the same, then joinedRow must be the same?
2. When the exist it ture?

Reply:
I don't remember the details of this part, but `exist` can be true. I tested it both in memory footprint numbers and in this way:
Let it panic when `exist` is true and run:
```sql
MySQL root@172.16.5.181:CREDIT_CARD> insert into t_3 select * from t_2 where vip_level=0 limit 200000
Query OK, 200000 rows affected
Time: 7.873s
MySQL root@172.16.5.181:CREDIT_CARD> insert into t_4 select * from t_3 limit 10
Query OK, 10 rows affected
Time: 2.484s
MySQL root@172.16.5.181:CREDIT_CARD> explain analyze delete t_3, t_4 from t_3, t_4 where `t_3`.`vip_level` = `t_4`.`vip_level`
(1105, 'duplicate handle!!')
```